lindi/LindiH5pyFile/LindiH5pyDataset.py

```diff
@@ -34,9 +36,19 @@ def __init__(self, _dataset_object: Union[h5py.Dataset, zarr.Array], _file: "Lin
         if isinstance(_dataset_object, zarr.Array):
             compound_dtype_obj = _dataset_object.attrs.get("_COMPOUND_DTYPE", None)
             if compound_dtype_obj is not None:
+                assert isinstance(compound_dtype_obj, list)
+                for i in range(len(compound_dtype_obj)):
```

Comment:
```suggestion
                # compound_dtype_obj is a list of tuples (name, dtype)
                # where dtype == "<REFERENCE>" if it represents an HDF5 reference
                for i in range(len(compound_dtype_obj)):
```

Adding comments as a reminder of the internal format